ground/gcs/src/plugins/scope/scopegadgetoptionspage.cpp

```diff
@@ -91,20 +91,36 @@ QWidget* ScopeGadgetOptionsPage::createPage(QWidget *parent)
     ExtensionSystem::PluginManager *pm = ExtensionSystem::PluginManager::instance();
     UAVObjectManager *objManager = pm->getObject<UAVObjectManager>();
     QVector< QVector<UAVDataObject*> > objList = objManager->getDataObjectsVector();
+    QStringList toAdd;
+
+    QString previous = NULL;
+
     foreach (QVector<UAVDataObject*> list, objList) {
         foreach (UAVDataObject* obj, list) {
+	    QString name = obj->getName();
+
             if (obj->isSingleInstance())
             {
-                options_page->cmbUAVObjects->addItem(obj->getName());
+		toAdd.append(name);
             }
-            else if(obj->getName() != options_page->cmbUAVObjects->itemText(options_page->cmbUAVObjects->count()-1))
+            else if(name != previous)
             { //Checks to see if we're duplicating UAVOs because of multiple instances
-                options_page->cmbUAVObjects->addItem(obj->getName());
-                options_page->cmbUAVObjectsSpectrogram->addItem(obj->getName());
+		toAdd.append(name);
             }
+
+            previous = name;
         }
     }
 
+    qSort(toAdd);
+
+    foreach (QString elem, toAdd) {
+	// Before we only added items to the spectrogram if they were not
+	// single instance; assuming that was in error
```

Comment:
FYI, we only added multi-instance items because each instance was one bin. So we should probably fix this so that the spectrogram only shows UAVO that it can plot, or teach it how to work with single-instance UAVOs (which would be pretty cool!)